cmd/monaco/download/downloaders.go

```diff
@@ -33,28 +36,92 @@ func makeDownloaders(options downloadConfigsOptions) (downloaders, error) {
 		return nil, err
 	}
 
-	var automationDownloader download.Downloader[v2.AutomationType] = dlautomation.NoopAutomationDownloader{}
+	var automationDownloader download.Downloader[config.AutomationType] = dlautomation.NoopAutomationDownloader{}
 	if clients.Automation() != nil {
 		automationDownloader = dlautomation.NewDownloader(clients.Automation())
 	}
-	var settingsDownloader download.Downloader[v2.SettingsType] = settings.NewDownloader(clients.Settings())
-	var classicDownloader download.Downloader[v2.ClassicApiType] = classic.NewDownloader(clients.Classic())
-	return downloaders{settingsDownloader, classicDownloader, automationDownloader}, nil
+	var settingsDownloader download.Downloader[config.SettingsType] = settings.NewDownloader(clients.Settings())
+	var classicDow download.Downloader[config.ClassicApiType]
+	if d, e := classicDownloader(clients.Classic(), options); e != nil {
+		classicDow = d
+	} else {
+		return nil, e
+	}
+	return downloaders{settingsDownloader, classicDow, automationDownloader}, nil
+}
+
+func classicDownloader(client dtclient.Client, opts downloadConfigsOptions) (*classic.Downloader, error) {
+	endpoints := prepareAPIs(opts)
+	return classic.NewDownloader(client, classic.WithEndpoints(endpoints))
+}
+
+func prepareAPIs(opts downloadConfigsOptions) api.APIs {
+	switch {
+	case opts.onlyAutomation:
+		return nil
+	case opts.onlySettings:
+		return nil
+	case opts.onlyAPIs:
+		return api.NewAPIs().Filter(skipDownloadFilter, removeDeprecated(withWarn()))
+	case len(opts.specificAPIs) > 0:
+		return api.NewAPIs().Filter(api.RetainByName(opts.specificAPIs), skipDownloadFilter, warnDeprecated())
+	case len(opts.specificSchemas) == 0:
+		return api.NewAPIs().Filter(skipDownloadFilter, removeDeprecated())
+	default:
+		return nil
+	}
 }
 
-func (d downloaders) Classic() download.Downloader[v2.ClassicApiType] {
-	return getDownloader[v2.ClassicApiType](d)
+func skipDownloadFilter(api api.API) bool {
+	if api.SkipDownload {
+		log.Info("API can not be downloaded and needs manual creation: '%v'.", api.ID)
+		return true
+	}
+	return false
+}
+
+func removeDeprecated(log ...func(api api.API)) api.Filter {
```

Comment:
The functional options seems a bit overkill for this usecase, especially as you only ever apply the first (line 87).

Couldn't you achieve the same effect with a simple `logWarning bool` parameter?

Reply:
Just to be more readable on the caller side: `api.NewAPIs().Filter(removeSkipDownload, removeDeprecated(withWarn()))`

As it is an implementation detail, I'll leave it like it is